### ISSUE TITLE
Clean up use of unittest assert functions in twisted/news

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8384]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8384|
|Type|enhancement|
|Created|2016-06-03 02:08:11Z|
|Branch|https://github.com/twisted/twisted/tree/8384-rodrigc-assert|

This ticket is split out from [ticket:8371] because the patch there was too big to review.

The files in twisted/news
should have their use of unittest assert functions cleaned up so that:

* the most appropriate unittest assert function is used appropriately in each file
* use of deprecated aliases should be eliminated, to reduce deprecation
  warnings and errors when moving to Python3 compatibility

This has some pointers about deprecated unittest assert aliases:

See: https://docs.python.org/2.7/library/unittest.html#deprecated-aliases

|= Method Name =|= Deprecated alias(es) =|
|---------------|------------------------|
| assertEqual() | failUnlessEqual, assertEquals |
| assertNotEqual() | failIfEqual |
| assertTrue() | failUnless, assert_ |
| assertFalse() | failIf |
| assertRaises() |	failUnlessRaises |
| assertAlmostEqual() | failUnlessAlmostEqual  |
| assertNotAlmostEqual() | failIfAlmostEqual  |

<details><summary>Searchable metadata</summary>

```
trac-id__8384 8384
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8384_rodrigc_assert 8384-rodrigc-assert
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1464919691052967 1464919691052967
changetime__1465772120576667 1465772120576667
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/100

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph set owner to @rodrigc|
|-|-|

Thanks for your contribution, rodrigc.

As with [#8376](https://github.com/twisted/twisted/issues/8376), the changes to these assertion functions are over-broad.  These are a lot of changes to look over, but please re-submit more minimal changes that just address the necessary bits to migrate to python 3.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc removed owner|
|-|-|

OK, I uploaded a new diff which just removes use of deprecated aliases:

https://github.com/twisted/twisted/pull/100

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set owner to @rodrigc|
|-|-|

Looks good. Thanks!

Waiting for build results and will merge.

Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset fd268d3a81679ecea415ba706352d8ae187748e8
```
#!CommitTicketReference repository="" revision="fd268d3a81679ecea415ba706352d8ae187748e8"
Merge 8384-rodrigc-assert: Clean up unittest assert functions in twisted/news.

Author: rodrigc
Reviewer: adiroiban
Fixes: #8384
```
